### DISCUSSION TITLE
mimxrt1060_evkb with rk043fn66hs_ctg shield issue

### DISCUSSION THREAD

**Discussion by castrorazo** [zephyrproject-rtos/zephyr#75688] [General] [answered]
Hello, I'm having issues with my shield not responding to any touch inputs, I tried running my project using the mimxrt1060_evkb board and rk043fn66hs_ctg touch shield as the target hardware. I also tried running the sample via:
    west build -b mimxrt1060_evkb samples/subsys/display/lvgl/ --pristine -- -DSHIELD=rk043fn66hs_ctg
which displayed graphics fine, but was unable to use the touch input. 
This is a continuation of a previous conversion: #75193 
I did use example code from mcu expresso to make sure my hardware was functioning properly, and the graphics/touch input worked fine when doing that. Wasn't sure if there was some type of setup I needed to do, or If the main branch isn't up to date for this target hardware. Any suggestions would be greatly appreciated.

**Comment by DerekSnell** [comment]
Hi @castrorazo ,
I tested your build command on my hardware, and it works great.  I used Zephyr v3.7.0-rc2, with your command:
```
west build -b mimxrt1060_evkb samples/subsys/display/lvgl/ --pristine -- -DSHIELD=rk043fn66hs_ctg
```
To rule out a build issue, attached is the binary image I used [zephyr_bin.zip](https://github.com/user-attachments/files/16164549/zephyr_bin.zip).  You can program this to address 0x60000000.

Otherwise, can you humor me and try building for the RK043FN02H-CT shield instead.  There are two display shields for these EVKs with similar part numbers, and they have the same display controller but different touch controllers. If you build with the wrong shield name, the display will still work but the touchscreen will not. The older shield is [RK043FN02H-CT](https://docs.zephyrproject.org/latest/boards/shields/rk043fn02h_ct/doc/index.html), which was replaced by the latest [RK043FN66HS-CTG](https://docs.zephyrproject.org/latest/boards/shields/rk043fn66hs_ctg/doc/index.html).

Let us know what you find.  Thanks

**Comment by castrorazo** [comment]
Hello,
First of all I would like to thank you for your response, I did not see this right away. I would like to note I'm just using the main Zephyr branch for development, not the v3.7.0-rc2 version. 
I did try flashing that binary image using: pyocd flash --target mimxrt1060 --base-address 0x60000000 ..../zephyr_bin.bin
Again I got the same outcome, the graphics work fine, but the input touch is not responsive. 
I did also try the older shield just to humor the idea, but as expected that did not do anything. I did however confirm on the back of my shield that I'm using the new up to date shield. Any other suggestions would be greatly appreciated.

**Comment by DerekSnell** [comment]
Hi @castrorazo ,
Thank you for trying this.  Since you have proven out your hardware works with an MCUXpresso SDK example, it seems there is some hardware difference impacting the Zephyr app.  What revision of the MIMXRT1060-EVKB board do you have?  There should be a sticker on the bottom of the board that looks like "SCH-xxxxx REV xx".  Can you fill in the `x`s and share the full label?

The touch controller is on an I2C bus.  And there is a GPIO interrupt signal the controller uses to indicate a touch event.  I suspect the issue is related to those interfaces.  If you are able to debug the code using those interfaces, or monitor those signals, it may help us identify the root cause.

Best regards

**Comment by castrorazo** [comment, marked as answer]
"SCH-47858 REV B1"
 
 I will take a look at the schematic and try to diagnose the gpio interrupt. Thank you!

**Comment by DerekSnell** [comment]
Hi @castrorazo ,
"SCH-47858 REV B1" is the same revision I have.  So it seems we will need to debug this issue on your hardware to better understand the problem.

Let us know what you find.  Thanks
